pypeit/core/telluric.py

```diff
@@ -1233,39 +1274,135 @@ def create_bal_mask(wave):
+        scipy.optimize.differential_evolution for details.
+
+    popsize (int): default = 30
+        A multiplier for setting the total population size for the differential evolution optimization. See
+        scipy.optimize.differential_evolution for details.
+
+    recombination : float, optional, default = 0.7
+        The recombination constant for the differential evolution optimization. This should be in the range [0, 1].
+        See scipy.optimize.differential_evolution for details.
+
+    polish : bool, optional, default=True
+        If True then differential evolution will perform an additional optimizatino at the end to polish the best fit
+        at the end, which can improve the optimization slightly. See scipy.optimize.differential_evolution for details.
+    disp : bool, optional, default=True
+        Argument for scipy.optimize.differential_evolution which will  display status messages to the screen
+        indicating the status of the optimization. See above for a description of the output and how to know
+        if things are working well.
+
+    debug_init : bool, optional, default=False
+        Show plots to the screen useful for debugging model initialization
+
+    debug : bool, optional, default=False
+        Show plots to the screen useful for debugging the telluric/object model fits.
+
+    Returns
+    -------
+    meta_table : astropy.table.Table object
+        Table containing meta data for the telluric/object model fit
+
+    out_table: astropy.table.Table. object
+        Table containing the values for telluric model parameters and object model parameters.
+    """
 
     # Turn on disp for the differential_evolution if debug mode is turned on.
     if debug:
         disp = True
 
     obj_params = dict(pca_file=pca_file, npca=npca, z_qso=z_qso, delta_zqso=delta_zqso, bounds_norm=bounds_norm,
                       tell_norm_thresh=tell_norm_thresh,
-                      output_meta_keys=('pca_file', 'npca', 'z_qso', 'delta_zqso','bounds_norm', 'tell_norm_thresh'))
+                      output_meta_keys=('pca_file', 'npca', 'z_qso', 'delta_zqso','bounds_norm', 'tell_norm_thresh'),
+                      debug_init=debug_init)
 
     wave, flux, ivar, mask, meta_spec, header = general_spec_reader(spec1dfile, ret_flam=True)
     header = fits.getheader(spec1dfile) # clean this up!
     # Mask the IGM and mask wavelengths that extend redward of our PCA
     qsomask = (wave > (1.0 + z_qso)*pca_lower) & (wave < pca_upper*(1.0 +
                                                                     z_qso))
     # TODO this 3100 is hard wired now, but make the QSO PCA a PypeIt product and determine it from the file
-    if create_bal_mask is not None:
-        bal_mask = create_bal_mask(wave)
-        mask_tot = mask & qsomask & bal_mask
+    if bal_mask is not None:
+        bal_gpm = create_bal_mask(wave, bal_mask)
+        mask_tot = mask & qsomask & bal_gpm
     else:
         mask_tot = mask & qsomask
```

Comment:
This can be made more compact, i.e. 

mask_tot = mask & qsomask & create_bal_mask(wave, bal_mask) if bal_mas is not None else mask & qsomask

Reply:
Thanks